### PR TITLE
Setup functional and smoke tests for bulk-scan-processor in aat

### PR DESCRIPTION
**Pull request by lgonczar**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-1229

### Change description ###

Setup functional and smoke tests for bulk-scan-processor in aat

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: k8s/aat/common/bsp/bulk-scan-processor.yaml [hunk 1]
```diff
@@ -25,6 +36,45 @@ spec:
...
+          PROXYOUT_ENABLED: true
+          RUN_END_TO_END_TESTS: true
+      smoketestscron:
+        enabled: true
+        image: hmctspublic.azurecr.io/bulk-scan/processor-test:prod-1795cc55
+        schedule: "0/30 * * * *"
+      functionaltestscron:
+        enabled: true
+        image: hmctspublic.azurecr.io/bulk-scan/processor-test:prod-1795cc55
+        schedule: "15/30 * * * *"
```
Comment (lgonczar) [inline k8s/aat/common/bsp/bulk-scan-processor.yaml:71]: These two will go away later, but this is to verify if scheduled execution works as expected.

File: k8s/aat/common/bsp/bulk-scan-processor.yaml [hunk 2]
```diff
@@ -25,6 +36,45 @@ spec:
...
+      smoketests:
+        enabled: true
+        image: hmctspublic.azurecr.io/bulk-scan/processor-test:prod-1795cc55
```
Comment (mdayican) [inline k8s/aat/common/bsp/bulk-scan-processor.yaml:74]: what is this 'processor-test' bit?

File: k8s/aat/common/bsp/bulk-scan-processor.yaml [hunk 3]
```diff
@@ -25,6 +36,45 @@ spec:
...
+            resourceGroup: bulk-scan
+            secrets:
+              storage-account-primary-key: TEST_STORAGE_ACCOUNT_KEY
+              storage-account-name: TEST_STORAGE_ACCOUNT_NAME
+              processed-envelopes-queue-send-connection-string: PROCESSED_ENVELOPES_QUEUE_WRITE_CONN_STRING
+          s2s:
+            resourceGroup: rpe-service-auth-provider-aat
+            secrets:
+              microservicekey-bulk-scan-processor-tests: TEST_S2S_SECRET
+        environment:
```
Comment (mdayican) [inline k8s/aat/common/bsp/bulk-scan-processor.yaml:53]: there 2 environments. one on the above

File: k8s/aat/common/bsp/bulk-scan-processor.yaml [hunk 4]
```diff
@@ -25,6 +36,45 @@ spec:
...
+          TEST_URL: "http://bulk-scan-processor-java"
+          SLACK_CHANNEL: "bsp-build-notices"
+          SLACK_NOTIFY_SUCCESS: true
+          TEST_S2S_NAME: "bulk_scan_processor_tests"
+          TEST_S2S_URL: "http://rpe-service-auth-provider-aat.service.core-compute-aat.internal"
+          TEST_SCAN_DELAY: "4000"
+          TEST_STORAGE_CONTAINER_NAME: "bulkscan"
```
Comment (mdayican) [inline k8s/aat/common/bsp/bulk-scan-processor.yaml:60]: do we need these new vars?

File: k8s/aat/common/bsp/bulk-scan-processor.yaml (hunk 3 above)
Comment (lgonczar) [reply k8s/aat/common/bsp/bulk-scan-processor.yaml:53]: One is for the service itself, this one is for tests.

File: k8s/aat/common/bsp/bulk-scan-processor.yaml (hunk 2 above)
Comment (lgonczar) [reply k8s/aat/common/bsp/bulk-scan-processor.yaml:74]: That's by convention:
```
hmctspublic.azurecr.io/{product}/{service}-test
```

File: k8s/aat/common/bsp/bulk-scan-processor.yaml (hunk 4 above)
Comment (lgonczar) [reply k8s/aat/common/bsp/bulk-scan-processor.yaml:60]: yes, they're used by tests

File: k8s/aat/common/bsp/bulk-scan-processor.yaml (hunk 3 above)
Comment (mdayican) [reply k8s/aat/common/bsp/bulk-scan-processor.yaml:53]: I see them in the same level?

File: k8s/aat/common/bsp/bulk-scan-processor.yaml (hunk 4 above)
Comment (lgonczar) [reply k8s/aat/common/bsp/bulk-scan-processor.yaml:60]: It's all under `testsConfig`

**Pull request merged**